components/script/script_thread.rs

```diff
@@ -480,6 +482,9 @@ pub struct ScriptThread {
 
     microtask_queue: MicrotaskQueue,
 
+    /// The unit of related similar-origin browsing contexts' list of MutationObserver objects
+    mutation_observers: Vec<JS<MutationObserver>>,
```

Comment:
This needs to be `DOMRefCell<Vec<JS<MutationObserver>>>` (which provides the borrow_mut() method that you're calling).